### PR TITLE
[TF FE] Added additional pruned inputs for MetaGraph support

### PR DESCRIPTION
**Pull request by gkrivor**
### Details:
 - Added handling of additional pruned inputs
 - Added possible topology of RestoreV2 -> AssignVariableOp
 - Added additional checks

### Tickets:
 - ...

### CODE REVIEW

File: src/frontends/tensorflow/src/input_model.cpp [hunk 1]
```diff
@@ -346,6 +347,7 @@ std::vector<std::shared_ptr<OpPlace>> InputModel::InputModelTFImpl::topologicall
                 if (m_tensor_places.find(producer_name) != m_tensor_places.end()) {
                     const auto& tensor_place = m_tensor_places[producer_name];
                     is_input |= tensor_place->is_input();
+                    m_found_inputs.insert(producer_name);
                 }
```
Comment (rkazants) [inline src/frontends/tensorflow/src/input_model.cpp:352]: please add test to cover this corner case

Comment (gkrivor) [reply src/frontends/tensorflow/src/input_model.cpp:352]: Do you know how? I have no ideas

Comment (gkrivor) [reply src/frontends/tensorflow/src/input_model.cpp:352]: Moreover, I think all three ways should be covered, but I don't know how to emulate this situations right now...

My suggestion: if this change should be merged before code freeze - create a task for tests and work on it separately. If I find a way how to test it earlier - add the test right here. @rkazants what do you think?

Comment (rkazants) [reply src/frontends/tensorflow/src/input_model.cpp:352]: Don't think it is complicated to simulate. Let us discuss this offline - I will tell you.

**Review by rkazants — CHANGES_REQUESTED** [review]
Please add tests to cover pruning by input and output ports for Meta format. We discussed these tests offline. Absolutely it is easy to implement.
This way we can disclose bugs if any remained.

File: src/frontends/tensorflow/src/variables_index.cpp [hunk 2]
```diff
@@ -448,8 +448,20 @@ void VariablesIndex::map_assignvariable(const std::shared_ptr<::tensorflow::Grap
 
             if (restorev2_nodes.size() == 1 && varhandle_nodes.size() == 1) {
                 std::vector<std::string> restore_output;
-                // Expected path is: RestoreV2 -(output_index)-(0)-> Identity -(0)-(1)-> AssignVariableOp
-                PtrNode::parse_node_name(node.second->inputs[1]->node->input(0), restore_output);
+
+                FRONT_END_GENERAL_CHECK(node.second->inputs.size() >= 2,
+                                        "Amount of AssignVariableOp inputs is less than expected");
+                // Here is known ways to find a correct RestoreV2 output index:
```
[the remark this answers is not in the document]
Comment (rkazants) [reply src/frontends/tensorflow/src/variables_index.cpp:454]: minor

**Review by rkazants — APPROVED** [review]

File: src/frontends/tensorflow/src/variables_index.cpp [hunk 3]
```diff
@@ -473,7 +485,10 @@ void VariablesIndex::map_assignvariable(const std::shared_ptr<::tensorflow::Grap
 
             if (restorev2_nodes.size() == 1 && variablev2_nodes.size() == 1) {
                 std::vector<std::string> restore_output;
-                // Expected path is: RestoreV2 -(output_index)-(0)-> Assign
+
+                FRONT_END_GENERAL_CHECK(node.second->node->input_size() >= 2,
+                                        "Amount of Assign inputs is less than expected");
+                // Expected path is: RestoreV2 -(output_index)-(1)-> Assign
                 PtrNode::parse_node_name(node.second->node->input(1), restore_output);
```
[the remark this answers is not in the document]
Comment (gkrivor) [reply src/frontends/tensorflow/src/variables_index.cpp:492]: No known models

[the remark this answers is not in the document]
Comment (rkazants) [reply src/frontends/tensorflow/src/variables_index.cpp:492]: Should we check check op_type of the node expected to be RestoreV2 as you did above?

[the remark this answers is not in the document]
Comment (rkazants) [reply src/frontends/tensorflow/src/variables_index.cpp:492]: @gkrivor, please fix it in the next PR

**Pull request merged**